src/views/CartView.vue

```diff
@@ -113,6 +130,40 @@ export default Vue.extend({
   },
   methods: {
     ...mapActions(['save']),
+    toggleAllVariables: async function (loadVariables) {
+      if (!loadVariables) {
+        this.$store.commit('updateGridSelection', {})
+        return
+      }
+
+      this.isLoadingallVariables = true
+      const attrs = 'id,name,label,variants(id,assessment_id)'
+      let fetchMore = true
+
+      const batch = { num: 1000, start: 0 }
+      const selection = {}
+
+      while (fetchMore) {
+        const response = await api.get(`/api/v2/lifelines_variable?&attrs=${attrs}&start=${batch.start}&num=${batch.num}&sort=id`)
+        batch.start += batch.num
+        if (response.items.length !== batch.num) {
+          fetchMore = false
+        }
+
+        const variables = response.items.map(toVariable)
+
+        for (const variable of variables) {
+          const variableAssessmentIds = transforms.gridAssessments(variable.variants, this.$store.state.assessments, null).map((i) => i.id)
+          if (variableAssessmentIds.length) {
+            selection[variable.id] = variableAssessmentIds
+          }
+        }
+      }
+
+      this.isLoadingallVariables = false
+
+      this.$store.commit('updateGridSelection', selection)
+    },
```

Comment:
to to actions and suggest to replace while loop with size check and number, any mistake while loop can cause browser to become unresponsive, and the length is known upfront

Reply:
This method is specific to the CartView, so I'll rather keep it here. The total amount of items is not known upfront; a first api request is needed to get response.total; that's why a while loop is being used.